source/jormungandr/jormungandr/interfaces/v1/EquipmentReports.py

```diff
@@ -84,12 +84,19 @@ def _create_filter_equipment(self, instance):
             abort(404, message='No code type exists into equipment provider')
         return "stop_point.has_code_type(" + code_types + ")"
 
-    def get(self, region=None, lon=None, lat=None):
+    def get(self, region=None, lon=None, lat=None, uri=None):
         self.region = i_manager.get_region(region, lon, lat)
         timezone.set_request_timezone(self.region)
         args = self.parsers["get"].parse_args()
         instance = i_manager.instances.get(self.region)
 
+        uris = []
+        if uri:
+            if uri[-1] == "/":
+                uri = uri[:-1]
+            uris = uri.split("/")
+        args["filter"] = self.get_filter(uris, args)
```

Comment:
Duplicated code spotted :) 
It would be so much nicer to extract this out in a separate function and use it across all classes that support filtering (LineReports, Schedules, Disruptions, Calendars...)

Reply:
If it 's already exist a common, I'll do it